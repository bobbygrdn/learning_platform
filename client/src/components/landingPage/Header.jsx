import React from 'react'
import '../../styles/landingPage/Header.css'

export default function Header() {
    return (
        <>
            <header className='header' data-testid="header">
                <div className='titles'>
                    <p className="title" data-testid="title">Valor Academy</p>
                    <p className="sectiontitle" data-testid="sectiontitle">Become a Code Warrior</p>
                </div>

                <img className="logo" src={process.env.PUBLIC_URL + '/resources/valorAcademy_logo.png'} data-testid="logo" alt="Valor Academy logo" />
            </header >
        </>
    )
}
