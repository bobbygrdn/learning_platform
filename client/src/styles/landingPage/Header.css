.header {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  pointer-events: none;
  margin-bottom: 1em;
}

.titles {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.title {
  margin-bottom: 0;
  height: fit-content;
}

.logo {
  aspect-ratio: 1;
  width: 9vw;
  margin-left: 1em;
}

@media screen and (max-width: 600px) {
  .logo {
    width: 16vw;
    height: 12vh;
  }
}

@media screen and (min-width: 1000px) {
  .title {
    font-size: 52px;
  }

  .sectiontitle {
    font-size: 28px;
  }
}
