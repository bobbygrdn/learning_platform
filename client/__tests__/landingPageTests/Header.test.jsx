import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from '../../src/components/landingPage/Header';

it('renders without crashing', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
})

it('should render header element', () => {
    render(<Header />);
    expect(screen.getAllByTestId('header')).toHaveLength(1);
})

it('should render title element', () => {
    render(<Header />);
    expect(screen.getAllByTestId('title')).toHaveLength(1);
})

it('should render sectiontitle element', () => {
    render(<Header />);
    expect(screen.getAllByTestId('sectiontitle')).toHaveLength(1);
})

it('should render logo element', () => {
    render(<Header />);
    expect(screen.getAllByTestId('logo')).toHaveLength(1);
})