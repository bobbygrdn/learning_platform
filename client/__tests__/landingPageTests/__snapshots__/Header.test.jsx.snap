// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <header
    class="header"
    data-testid="header"
  >
    <div
      class="titles"
    >
      <p
        class="title"
        data-testid="title"
      >
        Valor Academy
      </p>
      <p
        class="sectiontitle"
        data-testid="sectiontitle"
      >
        Become a Code Warrior
      </p>
    </div>
    <img
      alt="Valor Academy logo"
      class="logo"
      data-testid="logo"
      src="undefined/resources/valorAcademy_logo.png"
    />
  </header>
</div>
`;
